src/Microsoft.AspNet.Razor.Runtime/TagHelpers/ITypeInfo.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    /// <summary>
+    /// Contains type metadata.
+    /// </summary>
+    public interface ITypeInfo : IMemberInfo
+    {
+        /// <summary>
+        /// Fully qualified name of the type.
+        /// </summary>
+        string FullName { get; }
+
+        /// <summary>
+        /// Gets <see cref="IPropertyInfo"/>s for all properties of the current type excluding indexers.
+        /// </summary>
+        IEnumerable<IPropertyInfo> Properties { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the type is public.
+        /// </summary>
+        bool IsPublic { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the type is abstract.
+        /// </summary>
+        bool IsAbstract { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the type is generic.
+        /// </summary>
+        bool IsGenericType { get; }
+
+        /// <summary>
+        /// Determines whether the current type implements the specified interface.
+        /// </summary>
+        /// <param name="interfaceType">The <see cref="TypeInfo"/> of the interface.</param>
+        /// <returns><c>true</c> if the type implements the interface, otherwise <c>false</c>.</returns>
+        bool ImplementsInterface(TypeInfo interfaceType);
```

Comment:
In precompilation, we might eventually be able to map compilation symbols to `Type`s (rather than looking them up by names). This would allow for such a case.